core/src/io/anuke/mindustry/input/InputHandler.java

```diff
@@ -418,17 +419,50 @@ protected void flushRequests(Array<BuildRequest> requests){
                 if(copy.hasConfig && copy.block.posConfig){
                     copy.config = Pos.get(Pos.x(copy.config) + copy.x - copy.originalX, Pos.y(copy.config) + copy.y - copy.originalY);
                 }
+
+                copy.block = didyoumean(req.x, req.y, req.block, req.rotation);
```

Comment:
Blocks should be swapped only when you are placing a line, not when you are flushing them to the main build queue. The player should be able to see that their line request blocks are turning to junctions before they choose to place them, without using any special hacks to get it to display that way in the rendering code. Also, this won't work with mobile input. Instead of changing code here, you should change `updateLine` to replace the build request blocks.